src/hooks/useWeb3React.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+import { ethers } from 'ethers';
+import { config } from '@/lib/config';
+import { TOKEN_ABI } from '@/constant/abis';
+
+export const useWalletBalances = (walletAddress: string) => {
+  const [ethBalance, setEthBalance] = useState<string>('0');
+  const [usdcBalance, setUsdcBalance] = useState<string>('0');
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchBalances = async () => {
+      try {
+        if (!walletAddress) {
```

Comment:
I assume this is to check for empty string (`''`). If so, you can be more descriptive and concise with:

```suggestion
        // Ignore empty string wallet address
        if (!walletAddress) return;
```

If you never expect this to be empty string and you are instead expecting and checking for `null` / `undefined`. Please type the hook parameter correctly: `(walletAddress?: string)`.